source/blog/2018-02-28-building-buildah-container-image-for-kubernetes.md

```diff
@@ -0,0 +1,147 @@
+
+Time to install Buildah. Run as root because you will need to be root for running Buildah commands for this exercise. My Linux of choice is Fedora and so I use DNF.
+
+```
+# dnf -y install buildah
+```
+
+Here are the steps I performed to build my Buildah OCI image:
+
+First I wanted to build it from scratch. I wanted nothing in the container except what I need to run Buildah. But what about DNF or Yum, you might ask? Don't need them. If I'm using this Buildah container from the command line then I'll be using the host’s package manager, whatever that might be. If I'm using Buildah `bud`, aka build-using-dockerfile, then I'm using the FROM images package manager. To start the process, I created a scratch container and stored the container's name, which happens to be working-container, in a environment variable for convenience.
+
+```  
+# containerid=$(buildah from scratch)
+```
+
+Now I need to mount the container's file system so that I can install the buildah package and dependencies.
+
+```
+# scratchmnt=$(buildah mount $containerid)
+```
+
+Next I install the buildah package and dependencies into the containers filesystem. Notice that as I'm doing this from a Fedora laptop I'm defaulting to the Fedora repositories and I'm specifying the version, 27. Also, I clean up afterward so that we can reduce the image size. If you skip the `dnf clean` step you'll have extra bloat.
+
+```
+# dnf install --installroot $scratchmnt --release 27 buildah --setopt install_weak_deps=false -y
+# dnf clean all --installroot $scratchmnt --release 27
+```
+
+Now we need to set the container ENTRYPOINT and CMD. It's good practice to use ENTRYPOINT for the command that you want to run. When containerizing a command line tool, I usually set CMD to `--help` so that it gets appended to the ENTRYPOINT if you don't specify any parameters. I got into the practice after reading Michael Crosby’s [Best Practices](http://crosbymichael.com/dockerfile-best-practices.html) post. See section 5. 
+
+ (Currently Buildah needs ‘\’ for parameters with ‘--’. That will get fixed.)
+
+```
+# buildah config --cmd "\-\-help" $containerid
+# buildah config --entrypoint /usr/bin/buildah $containerid
+```
+
+It’s also good practice, and pretty important, to set some other metadata like the image name and who created it etc.
+
+```
+# buildah config --author "wgh @ redhat.com" --created-by "ipbabble" --label name=buildah $containerid
+```
+
+Finally I need to commit the container to an image. It should be called `buildah` because it is good practice to name a command line tool container the name of the command. That way, if its ENTRYPOINT is set correctly, you can run it similar to the command it is containerizing. The following command ran a little over a minute and currently has not status output. So be patient.
```

Comment:
"not status output" to "no status output"

Reply:
+1